packages/GraphQLDataProvider/src/graphQLDataProvider.ts

```diff
@@ -901,8 +901,15 @@ export class GraphQLDataProvider extends ProviderBase implements IEntityDataProv
             if (EntityRelationshipsToLoad.indexOf(entityInfo.RelatedEntities[i].RelatedEntity) >= 0) {
                 const r = entityInfo.RelatedEntities[i];
                 const re = this.Entities.find(e => e.ID === r.RelatedEntityID);
+                let uniqueCodeName: string = '';
+                if (r.Type.toLowerCase().trim() === 'many to many') {
+                    uniqueCodeName = `${r.RelatedEntityCodeName}_${r.JoinEntityJoinField.replace(/ /g, '')}`;
+                }
+                else {
+                    uniqueCodeName = `${r.RelatedEntityCodeName}_${r.RelatedEntityJoinField.replace(/ /g, '')}`;
```

Comment:
Not sure if we'll ever see tabs, but it doesn't hurt to collapse all whitespace

```suggestion
                if (r.Type.toLowerCase().trim() === 'many to many') {
                    uniqueCodeName = `${r.RelatedEntityCodeName}_${r.JoinEntityJoinField.replace(/\s/g, '')}`;
                }
                else {
                    uniqueCodeName = `${r.RelatedEntityCodeName}_${r.RelatedEntityJoinField.replace(/\s/g, '')}`;
```